### PR TITLE
Add onStatusBarTapped event callback on Scaffold

### LINKED ISSUE #85603: ios status bar touch scroll top move doesn't work in nested navigation.
ios status bar touch scroll top move doesn't work in nested navigation. 

Is there any workaround for this?

Nested navigation is used a lot in many apps such as Netflix.

With one thought, I tried the following method, but it returns null and it doesn't work.

```dart
SingleChildScrollView(
      controller: PrimaryScrollController.of(Navigator.of(context, rootNavigator: true).context),
      ...
      ...
      ...
    )
```

### PR DESCRIPTION
**Pull request by SuhwanCha**
When you tap the status bar on iOS, the PrimaryScrollController scrolls to the top. However, when a Scaffold contains multiple scrollable widget, we cannot controller which scroll controller to animated.

This pull request introduces an onTapBarTapped callback function in the Scaffold. With this callback, you can customize the behavior when the status bar is tapped, such as scrolling to the top with your specific scroll controller or performing other actions.

@Piinks 
As we discussed on Discord, we don't need to implement this feature for Android. 
MitchellGoodwin said we need to test it works properly, but we don't need to add test case because this test already implemented on https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/scaffold_test.dart#L510

And we need to followed up PR that separate `Navigator` and `PrimaryScrollController`, but we still need this feature (onStatusBarTapped). Because if there are multiple scrollable widgets (i.e. TabBarView, IndexedStack), develop may want to implement event.

Followed issue #150505

Fixes #85603

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by SuhwanCha** [comment]
Hi @Piinks , I would appreciate it if you could let me know how do you think about this idea!

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -1927,6 +1928,12 @@ class Scaffold extends StatefulWidget {
   ///    Flutter.
   final String? restorationId;
 
+
+  /// Called when the status bar is tapped. (iOS only)
```
Comment (Piinks) [inline packages/flutter/lib/src/material/scaffold.dart:1932]: This does not feel appropriate as iOS only. Other devices have a status bar no?

File: AUTHORS [hunk 2]
```diff
@@ -124,3 +124,4 @@ Om Phatak <everythingoutdated@gmail.com>
 Amir Panahandeh <amirpanahandeh@gmail.com>
 Kostiantyn Sokolovskyi <sokolovskyi.konstantin@gmail.com>
 Valentin Vignal <valentin.vignal.dev@outlook.fr>
+Suhwan Cha <suhw4n@gmail.com>
```
Comment (Piinks) [inline AUTHORS:127]: There should be a newline here

File: packages/flutter/lib/src/material/scaffold.dart (hunk 1 above)
Comment (SuhwanCha) [reply packages/flutter/lib/src/material/scaffold.dart:1932]: Other devices have status bar, but only iOS supported that gesture. Then, how about rename this event to `oniOSStatusBarTapped`?

I  just wanted to make a name that would be unified with private method below

```
  // iOS FEATURES - status bar tap, back gesture

  // On iOS, tapping the status bar scrolls the app's primary scrollable to the
  // top. We implement this by looking up the primary scroll controller and
  // scrolling it to the top when tapped.
void _handleStatusBarTap() {
```

Comment (Piinks) [reply packages/flutter/lib/src/material/scaffold.dart:1932]: Other devices have status bars though, right? Why limit this to iOS?

**Comment by Piinks** [comment]
Hey @SuhwanCha would you like to continue working on this change?

**Pull request closed without merging**